test/clj/rems/application/test_approver_bot.clj

```diff
@@ -67,12 +63,13 @@
                :time (time/now)
                :application-id 1
                :comment ""}]
-             (generate-commands [created-event submitted-event] injections))))
+             (generate-commands submitted-event (apply-events [created-event submitted-event] injections)))))
 
     (testing "does not approve if the applicant is blacklisted for one of the resources"
       (let [injections (assoc injections :blacklisted? (fn [user resource]
-                                                         (= ["applicant" "urn:21"] [user resource])))]
-        (is (empty? (generate-commands [created-event submitted-event] injections)))))
+                                                         (= ["applicant" "urn:21"] [user resource])))
+            application (apply-events [created-event submitted-event] injections)]
+        (is (empty? (generate-commands submitted-event application)))))
```

Comment:
This pattern of building the application and passing the last event to generate-commands is duplicated many times in this test, so I'd extract a test helper for it (similar to the old generate-commands helper).

Reply:
I'd rather keep it explicit for now

Reply:
Ok